File: frontend/classes/ActionLog.ts
interface ActionLogData {
  id: number
  index: number
  step: number
  type: string
  faction: number | null
  data: string
  creation_date: string
}

class ActionLog {
  id: number
  index: number
  step: number
  type: string
  faction: number | null
  data: any
  creation_date: Date

  constructor(data: ActionLogData) {
    this.id = data.id
    this.index = data.index
    this.step = data.step
    this.type = data.type
    this.faction = data.faction
    this.data = data.data
    this.creation_date = new Date(data.creation_date)
  }
}

export default ActionLog

File: backend/rorapp/serializers/action_log.py
from rest_framework import serializers
from rorapp.models import ActionLog


# Serializer used to read action_logs
class ActionLogSerializer(serializers.ModelSerializer):
    class Meta:
        model = ActionLog
        fields = ("id", "index", "step", "type", "faction", "data", "creation_date")
